packages/core/src/components/forms/numericStepper.tsx

```diff
@@ -0,0 +1,255 @@
+
+        const classes = classNames(Classes.CONTROL_GROUP, { [Classes.DISABLED]: this.props.disabled }, className);
+
+        return <div className={classes}>{elems}</div>;
+    }
+
+    public componentDidUpdate() {
+        if (this.state.shouldSelectAfterUpdate) {
+            this.input.setSelectionRange(0, this.state.value.length);
+        }
+    }
+
+    private renderInputGroup() {
+        return (
+            <InputGroup
+                {...removeNonHTMLProps(this.props)}
+                intent={this.props.intent}
+                inputRef={this.inputRef}
+                key="input-group"
+                onChange={this.handleInputChange}
+                onKeyDown={this.handleKeyDown}
+                value={this.state.value}
+            />
+        );
+    }
+
+    private renderButton(
+            key: string,
+            iconName: string,
+            onClick: React.EventHandler<React.MouseEvent<HTMLButtonElement>>) {
+        return (
+            <Button
+                disabled={this.props.disabled || this.props.readOnly}
+                iconName={iconName}
+                intent={this.props.intent}
+                key={key}
+                onClick={onClick}
+            />
+        );
+    }
+
+    private inputRef = (input: HTMLInputElement) => {
+        this.input = input;
+    }
+
+    private handleDecrementButtonClick = (e: React.MouseEvent<HTMLInputElement>) => {
+        const direction = -1;
+        this.updateValue(direction, e.altKey, e.shiftKey);
+    }
+
+    private handleIncrementButtonClick = (e: React.MouseEvent<HTMLInputElement>) => {
+        const direction = 1;
+        this.updateValue(direction, e.altKey, e.shiftKey);
+    }
+
+    private handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+        if (this.props.disabled || this.props.readOnly) {
+            return;
+        }
+
+        const { keyCode } = e;
+
+        let direction: number;
+
+        if (keyCode === Keys.ARROW_UP) {
+            direction = 1;
+        } else if (keyCode === Keys.ARROW_DOWN) {
+            direction = -1;
+        } else {
+            return;
+        }
+
+        // we'd like to select the field contents after running the code in this
+        // onKeyDown handler, as a UX nicety. without e.preventDefault, some
+        // hotkeys (e.g. shift + up/down, alt + up/down) will clear the selection,
+        // resulting in an inconsistent or unintuitive experience.
+        e.preventDefault();
+
+        this.updateValue(direction, e.altKey, e.shiftKey);
+    }
+
+    private handleInputChange = (e: React.KeyboardEvent<HTMLInputElement>) => {
+        // the type definition for EventTarget (e.target) doesn't include
+        // the 'value' property out of the box, hence the janky casting
+        const nextValue = (e.target as any).value;
+        this.setState({ shouldSelectAfterUpdate : false, value: nextValue });
+
+        if (this.props.onChange) {
+            this.props.onChange(e);
```

Comment:
use `safeInvoke` for this pattern, from common/utils. [usage examples here](https://github.com/palantir/blueprint/search?utf8=%E2%9C%93&q=safeinvoke)